### PR TITLE
pretty_parse fails when code is not linted. Add option to open file

### PR DESCRIPTION
**Pull request by yogat3ch**
`pretty_parse` fails when there is any error in a code chunk (open parentheses etc) 
The fail error is often obtuse and doesn't provide much context. 
Added a function that catches the error and offers to open the file to the offending line (if applicable)
pretty_utils:
 - add `parse_check` internal function to handle a try error from `parse` in `pretty_parse`
 - add `parse_check` to `pretty_parse`
inst/Test.R:
 - This is a file to test the functionality of this feature as follows:
`pretty_namespace("inst/Test.R")`

### CODE REVIEW

File: R/pretty_utils.R [hunk 1]
```diff
@@ -1,7 +1,52 @@
-pretty_parse <- function(txt){
+#' @title parse_check
+#' @description check for fail of pretty_parse > parse, and offers to open file to offending line
+#' @keywords Internal
+#' @param p result of `pretty_parse` > `parse`
+#' @param txt input text to `pretty_parse` 
+#' @inheritParams pretty_namespace
+#' @importFrom rstudioapi navigateToFile
+#' @importFrom utils::askYesNo
```
Comment (yonicd) [inline R/pretty_utils.R:8]: typo in the importFrom call

File: R/pretty_utils.R [hunk 2]
```diff
@@ -1,7 +1,52 @@
...
+    # get the connection/input filename
+    .path <- get0(.vars[.var], envir = .sf[top_call][[1]], mode = "character")
+    if (interactive() && !is.null(.path)) {
+      if (!file.exists(.path)) stop(p)
+      # get the row & column
+      .rc <- as.numeric(strsplit(attr(p, "condition")$message, "\\:")[[1]][2:3])
+      # get the line number corresponding to the first line of text & add the rows indicated by the error (may not always be accurate but should work)
+      .line <- grep(txt[1], readLines(.path), fixed = TRUE) + .rc[1]
+      # Ask if the user wants to go to this line
+      .answer <- utils::askYesNo(paste0("Parse failed at line(s) ", paste0(.line, collapse = ", "),". Rstudio Users: Would you like to open the file at the line(s) before the function quits to correct the error (Rstudio Only)?"))
```
Comment (yonicd) [inline R/pretty_utils.R:32]: super long sentence that is not easy to digest in the conole. i think adding some line breaks will help

```r
      .answer <- utils::askYesNo(paste0("Parse failed at line(s) ", paste0(.line, collapse = ", "),". \nWould you like to open the file at the line(s) before\nthe function quits to correct the error (Rstudio Only)?"))
```

File: R/pretty_utils.R (hunk 1 above)
Comment (yogat3ch) [reply R/pretty_utils.R:8]: Oops, will fix.

File: R/pretty_utils.R (hunk 2 above)
Comment (yogat3ch) [reply R/pretty_utils.R:32]: Ok, shortened it. Commit coming soon.

**Review by yogat3ch — COMMENTED** [review]
Will make changes in upcoming commit

**Comment by yonicd** [comment]
looks good. ty!

**Comment by yogat3ch** [comment]
Certainly, thanks for merging all of these!

**Pull request merged**